truth_set/java/structure/organisation/consistency/Supermarket.java

```diff
@@ -0,0 +1,61 @@
+import java.util.Map;
+import java.util.HashMap;
+
+class UserService {
+    public String fetchUserProfile(int userId) {
+        try {
+            Thread.sleep(1000);
+        } catch (InterruptedException e) {
+            e.printStackTrace();
+        }
+        return "User Profile for ID: " + userId;
+    }
+
+    public boolean checkUserEligibility(int age) {
+        return age >= 18;
+    }
+}
+
+class ProductService {
+    public String getProduct(int productId) {
+        Map<Integer, String> products = new HashMap<>();
+        products.put(101, "Laptop");
+        products.put(102, "Smartphone"); 
+        products.put(103, "Headphones");
+        return products.getOrDefault(productId, "Product Not Found");
+    }
```

Comment:
<div>


<div id="suggestion">
<div id="issue"><b>Consider moving hardcoded data to config</b></div>
<div id="fix">

Consider moving the hardcoded data initialization to a separate configuration or data store. Having hardcoded values in the service classes makes maintenance difficult and violates separation of concerns.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<blockquote>Check the AI-generated fix before applying</blockquote>
<div id="code">

```
 @@ -20,6 +20,7 @@
      public String getProduct(int productId) {
 -        Map<Integer, String> products = new HashMap<>();
 -        products.put(101, "Laptop");
 -        products.put(102, "Smartphone");
 -        products.put(103, "Headphones");
 -        return products.getOrDefault(productId, "Product Not Found");
 +        return ProductDataStore.getProduct(productId);
      }
```


```suggestion
    public String getProduct(int productId) {
        return ProductDataStore.getProduct(productId);
    }
```



</div>
</details>
</div>


<small><i>Code Review Run <a href=https://github.com/rkj6124/bito-test/pull/3#issuecomment-2522127787>#dcd59f</a></i></small>
</div>

---
Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged